packages/react/src/Composite/Root/CompositeRoot.tsx

```diff
@@ -74,9 +76,14 @@ namespace CompositeRoot {
     itemSizes?: Dimensions[];
     dense?: boolean;
     enableHomeAndEndKeys?: boolean;
+    onMapChange?: (newMap: Map<Node, CompositeMetadata<Metadata> | null>) => void;
+    stopEventPropagation?: boolean;
+    rootRef?: React.RefObject<HTMLElement | null>;
```

Comment:
@michaldudak in the end James suggested using a custom ref prop to keep the types simple as this is an internal component

Reply:
That's pretty non-standard for our codebase. The types aren't that much complex, so my preference would be to use `forwardRef`, but I'll be able ti live with this approach as well. Your call.